### DISCUSSION TITLE
Getting rid of viewmodels as a parameter from Screens

### DISCUSSION THREAD

**Discussion by atimadanoglu** [waseefakhtar/dose-android#123] [General]
I think we should get rid of viewmodels as a parameter from Screen level. We should inject viewModels on Route level because when we try to preview the composables, it troubles us. Also, it creates other different problems. 

From Documentation: [https://developer.android.com/jetpack/compose/tooling/previews#preview-viewmodel](url) 
**Warning:** You shouldn't pass the ViewModel instances down to other composable functions. When you do this, you're coupling the composables with the ViewModel type, which makes them less reusable, harder to test, and unable to preview.

What do you think about that?

**Comment by waseefakhtar** [comment]
I think that's a better approach and we should go for it!

**Reply by atimadanoglu** [reply]
Can you open a task for this and assign to me?
